in_toto/models/link.py

```diff
@@ -27,6 +27,8 @@
 FILENAME_FORMAT_SHORT = "{step_name}.link"
 UNFINISHED_FILENAME_FORMAT = ".{step_name}.{keyid:.8}.link-unfinished"
 UNFINISHED_FILENAME_FORMAT_GLOB = ".{step_name}.{pattern}.link-unfinished"
+UNFINISHED_FILENAME_FORMAT_GLOB_DIR = \
+  "{metadata_directory}/.{step_name}.{pattern}.link-unfinished"
```

Comment:
I'm still not convinced by this.

- I think `glob` is smart about `//` in the overall pattern (this applies if the `metadata_directory` value includes a trailing `/`) but it's probably best avoided?
- We'll probably not catch permission issues.

Imagine a scenario where the metadata directory can be written to, but not read from, for whatever reason. `glob.glob` will search for the pattern with the `metadata_directory` included, but won't detect a lack of permission (we're not going to see `PermissionError`). It'll just return an empty list, and the user will get a message saying they didn't run `start`.

This all makes me wonder if we should even use `metadata_directory` for `in-toto-record start`. We continue the current policy of writing the unfinished link to the current working directory, and write only final metadata to the specified target in `in-toto-record stop`. Thoughts @lukpueh?

Reply:
I also agree with you about the idea that we may not need to use `metadata_director` for the `in-toto-record start`. Since we generate the final link when we run `in-toto-record stop`, maybe it will be more reasonable to decide the path to store the link at that time?

> I'm still not convinced by this.
> 
> * I think `glob` is smart about `//` in the overall pattern (this applies if the `metadata_directory` value includes a trailing `/`) but it's probably best avoided?
> * We'll probably not catch permission issues.
> 
> Imagine a scenario where the metadata directory can be written to, but not read from, for whatever reason. `glob.glob` will search for the pattern with the `metadata_directory` included, but won't detect a lack of permission (we're not going to see `PermissionError`). It'll just return an empty list, and the user will get a message saying they didn't run `start`.
> 
> This all makes me wonder if we should even use `metadata_directory` for `in-toto-record start`. We continue the current policy of writing the unfinished link to the current working directory, and write only final metadata to the specified target in `in-toto-record stop`. Thoughts @lukpueh?

Reply:
I also agree with you about the idea that we may not need to use metadata_director for the in-toto-record start. Since we generate the final link when we run in-toto-record stop, maybe it will be more reasonable to decide the path to store the link at that time?

Reply:
Yep, I'm leaning towards that. Let's get some opinions. :smile: